File: 1.0/src/c/cz/mg/vulkan/PFN_vkDestroyImageView.c
#include <jni.h>
#include <stdlib.h>
#include <string.h>
#include <vulkan/vulkan.h>

jlong jniPointerToLong(const void* p);
void* jniLongToPointer(jlong l);
jlong jniFunctionPointerToLong(PFN_vkVoidFunction p);
PFN_vkVoidFunction jniLongToFunctionPointer(jlong l);
void jniThrowException(JNIEnv* env, const char* message);

void Java_cz_mg_vulkan_PFNvkDestroyImageView_call(JNIEnv* env, jclass jc, jlong address, jlong device, jlong imageView, jlong pAllocator){
    (void)env;
    (void)jc;
    PFN_vkDestroyImageView f = (PFN_vkDestroyImageView)jniLongToFunctionPointer(address);
    f(
        *((VkDevice*)jniLongToPointer(device)),
        *((VkImageView*)jniLongToPointer(imageView)),
        ((VkAllocationCallbacks*)jniLongToPointer(pAllocator))
    );
}

File: 1.0/src/cz/mg/vulkan/PFNvkDestroyImageView.java
package cz.mg.vulkan;

public class PFNvkDestroyImageView extends VkFunctionPointer {
    public PFNvkDestroyImageView() {
    }

    protected PFNvkDestroyImageView(VkMemory vkmemory) {
        super(vkmemory);
    }

    protected PFNvkDestroyImageView(VkMemory vkmemory, long vkaddress) {
        super(vkmemory, vkaddress);
    }



    public PFNvkDestroyImageView(long value) {
        setValue(value);
    }

    public PFNvkDestroyImageView(VkInstance instance) {
        super(instance, new VkString("vkDestroyImageView"));
    }

    public void call(VkDevice device, VkImageView imageView, VkAllocationCallbacks pAllocator){
        callNative(getValue(), device != null ? device.getVkAddress() : VkPointer.getNullAddressNative(), imageView != null ? imageView.getVkAddress() : VkPointer.getNullAddressNative(), pAllocator != null ? pAllocator.getVkAddress() : VkPointer.NULL);
    }

    protected static native void callNative(long vkaddress, long device, long imageView, long pAllocator);

}
